sisyphus/simple_linux_utility_for_resource_management_engine.py

```diff
@@ -307,6 +316,10 @@ def queue_state(self):
         while True:
             try:
                 out, err, retval = self.system_call(system_command)
+                if retval != 0:
+                    logging.warning(self._system_call_error_warn_msg(system_command))
+                    time.sleep(gs.WAIT_PERIOD_BETWEEN_CHECKS)
+                    continue
```

Comment:
Not sure if this solves the problem.. If there is some fundamental problem with the queue, that is not recoverable, then this sends the function into an endless loop.

maybe instead just return the old cache?
```suggestion
                if retval != 0:
                    return self._task_info_cache
```
In the worst case we will never update the cache and a running/finished Job will still be marked as pending.

Reply:
IMHO I would log and wait, so that the normal cache update process can run

Reply:
define a new variable?